mantidproject/mantid
# Re #20991: Updated Reflectometry IDFs.

Updates the IDFs for INTER and SURF.
Removes `ReflectometryReductionOneAuto` v1
This also fixes #20991.

**To test:**
- In the ISIS Reflectometry interface, try autoreducing some `INTER` investigations dated both before and after the date the IDF changed (14/2/2017 for INTER), e.g. before: `1620396`, after: `1620384`. In v3.11 the latter gave an error about slit gaps not being found. You don't need to wait for all of the runs to reduce, as long as the first few groups process ok then the correct IDF is being picked up.
- Repeat the test for the following `SURF` investigations: before: `1610531`, after: `1700019`
- Run this script. The output for the nearest pixels for INTER should be around twoTheta=2.6 rather than theta=1.3. (Note that the output is not exact because the pixels are not centred exactly at twoTheta.) For SURF the output should be around twoTheta=3.04.
```python
import math
def getTwoThetaForPixel(wkspace, detId):
    inst = wkspace.getInstrument()
    det = inst.getDetector(detId)
    return wkspace.detectorTwoTheta(det) * 180 / math.pi
def doCheck(run, pixel):
    print run, "(nearest pixel is ", pixel, "):"
    ws=Load(Filename=run)
    detTwoTheta = getTwoThetaForPixel(ws,pixel)
    print '  Detector ID',pixel,'twoTheta =', detTwoTheta

print "INTER: Expect twoTheta to be around 2.6 in both cases"
doCheck("INTER43572", 4)
doCheck("INTER43583", 2047)
print
print "SURF: Expect twoTheta to be around 3.04"
doCheck("SURF121400", 30001)
```

Expected output:
```
INTER: Expect twoTheta to be around 2.6 in both cases
INTER43572 (nearest pixel is  4 ):
  Detector ID 4 twoTheta = 2.63604652684
INTER43583 (nearest pixel is  2047 ):
  Detector ID 2047 twoTheta = 2.59952111409

SURF: Expect twoTheta to be around 3.04
SURF121400 (nearest pixel is  30001 ):
  Detector ID 30001 twoTheta = 3.03571136117
```



Fixes #20991. 

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
